facade/src/main/java/com/redhat/lightblue/migrator/facade/ServiceFacade.java

```diff
@@ -94,6 +105,21 @@ public ServiceFacade(D legacySvc, D lightblueSvc, String implementationName, Pro
 
         timeoutConfiguration = new TimeoutConfiguration(DEFAULT_TIMEOUT_MS, implementationName, this.properties);
 
+        // create ThreadPoolExecutor for all facades
+        synchronized("executor") {
```

Comment:
For that to work, declare executor as volatile.

On Tue, May 17, 2016 at 12:52 PM, Naveen Malik <notifications@github.com>
wrote:

> In
> facade/src/main/java/com/redhat/lightblue/migrator/facade/ServiceFacade.java
> <https://github.com/lightblue-platform/lightblue-migrator/pull/350#discussion_r63581886>
> :
>
> > @@ -94,6 +105,21 @@ public ServiceFacade(D legacySvc, D lightblueSvc, String implementationName, Pro
> >
> >          timeoutConfiguration = new TimeoutConfiguration(DEFAULT_TIMEOUT_MS, implementationName, this.properties);
> >
> > +        // create ThreadPoolExecutor for all facades
> > +        synchronized("executor") {
>
> Actually, what about doing a check if executor is null first? This way you
> don't block on synchronization if it is is already set.
>
> if (executor == null) {
>    synchronized (CONFIG_PREFIX) {
>       if (executor == null) {
>          ...
>       }
>    }
> }
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-migrator/pull/350/files/cd061344fe4b5bf214de4bb196696bdbad982975#r63581886>
>